### PR TITLE
`cylc play`: add suggestion for `--upgrade` in non-interactive terminal

### PR DESCRIPTION
**Pull request by wxtim**
Companion to https://github.com/cylc/cylc-uiserver/pull/455

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] ~Tests are included (or explain why tests are not needed).~ One line message for Cylc GUI to pick up.
- [x] ~`CHANGES.md` entry included if this is a change that can affect users~ Covered by CHANGES.md in https://github.com/cylc/cylc-uiserver/pull/455
- [x] ~[Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.~ N/A
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/scheduler_cli.py [hunk 1]
```diff
@@ -523,6 +523,7 @@ def _version_check(
                     process=str.lower,
                 )
             # we are in non-interactive mode, abort abort abort
+            print(cparse('Use "--upgrade" to upgrade the workflow.'))
```
Comment (oliver-sanders) [inline cylc/flow/scheduler_cli.py:526]: * This message should not appear in interactive mode (covered by the prompts above).
* No need for `cparse` unless ANSI colour is used in the message.

```suggestion
            else:
                print('Use "--upgrade" to upgrade the workflow.')
```

File: cylc/flow/scheduler_cli.py [hunk 2]
```diff
@@ -523,6 +523,8 @@ def _version_check(
                     process=str.lower,
                 )
             # we are in non-interactive mode, abort abort abort
+            else:
+                print('Use "--upgrade" to upgrade the workflow.', file=sys.stderr)
```
Comment (wxtim) [inline cylc/flow/scheduler_cli.py:527]:
```suggestion
            print('Use "--upgrade" to upgrade the workflow.', file=sys.stderr)
```

Comment (wxtim) [reply cylc/flow/scheduler_cli.py]: The else isn't actually necessary, since the if contains a return.

File: cylc/flow/scheduler_cli.py [hunk 3]
```diff
@@ -523,6 +523,7 @@ def _version_check(
                     process=str.lower,
                 )
             # we are in non-interactive mode, abort abort abort
+            print('Use "--upgrade" to upgrade the workflow.', file=sys.stderr)
```
Comment (MetRonnie) [inline cylc/flow/scheduler_cli.py:526]: There is a bit of a mismatch printing this to stderr when just above
```
This workflow was previously run with <yellow>{last_run_version}</yellow>.
This version of Cylc is <green>{__version__}</green>.
```
was printed to stdout

Comment (wxtim) [reply cylc/flow/scheduler_cli.py:526]: @oliver-sanders  - Any reason not to print the earlier msg to stderr?

Comment (MetRonnie) [reply cylc/flow/scheduler_cli.py:526]: Unfortunately the GUI and TUI are only regurgitating stderr. We should find a way for them to include the useful info about which versions are flying about IMHO

Comment (oliver-sanders) [reply cylc/flow/scheduler_cli.py:526]: Fair point, but a widespread issue not related to this PR.

I don't think stdout/err are a great interface for Tui/GUI, but short of a Python API (that allows Cylc version changing) I can't think of a better option that isn't tons of work.

In this case, the version compat issue *should* also be handled better in at the UI end as we intend to implement upgrade logic there.

**Review by oliver-sanders — APPROVED** [review]

File: cylc/flow/scheduler_cli.py (hunk 3 above)
Comment (oliver-sanders) [reply cylc/flow/scheduler_cli.py:526]: Moved to #5554

**Pull request merged**